rasa/nlu/extractors/extractor.py

```diff
@@ -21,6 +21,90 @@ def add_processor_name(self, entity: Dict[Text, Any]) -> Dict[Text, Any]:
 
         return entity
 
+    def clean_up_entities(
+        self, entities: List[Dict[Text, Any]], keep: bool = True
+    ) -> List[Dict[Text, Any]]:
+        """
+        Checks if multiple entity labels are assigned to one word.
+
+        This might happen if you are using a tokenizer that splits up words into
+        sub-words and different entity labels are assigned to the individual sub-words.
+        In such a case keep the entity label with the highest confidence as entity
+        label for that word. If you set 'keep' to 'False', all entity labels for
+        that word will be removed.
+
+        Args:
+            entities: list of entities
+            keep:
+                If set to 'True', the entity label with the highest confidence is kept
+                if multiple entity labels are assigned to one word. If set to 'False'
+                all entity labels for that word will be removed.
+
+        Returns: updated list of entities
+        """
+        if len(entities) <= 1:
+            return entities
+
+        entity_indices: List[List[int]] = []
+
+        # get indices of entity labels that belong to one word
+        for idx in range(1, len(entities)):
+            if entities[idx]["start"] == entities[idx - 1]["end"]:
+                if entity_indices and entity_indices[-1][-1] == idx - 1:
+                    entity_indices[-1].append(idx)
+                else:
+                    entity_indices.append([idx - 1, idx])
+
+        entities_to_remove = set()
+
+        for indices in entity_indices:
+            if not keep:
+                entities_to_remove.update(indices)
+                continue
+
+            start = entities[indices[0]]["start"]
+            end = entities[indices[-1]]["end"]
+            value = "".join(entities[idx]["value"] for idx in indices)
+            idx = self._get_highest_confidence_idx(entities, indices)
+
+            if idx is None:
+                entities_to_remove.update(indices)
+            else:
+                indices.remove(idx)
+                entities_to_remove.update(indices)
+                entities[idx]["start"] = start
+                entities[idx]["end"] = end
+                entities[idx]["value"] = value
+
+        entities_to_remove = sorted(entities_to_remove, reverse=True)
```

Comment:
Why do we need this sorting? In general can we add a couple of comments to the whole procedure used here? It's a bit difficult to understand at first glance 😄